tedana/metrics/kundu_fit.py

```diff
@@ -222,20 +222,33 @@ def dependence_metrics(catd, tsoc, mmix, adaptive_mask, tes, ref_img,
 
     if verbose:
         # Echo-specific weight maps for each of the ICA components.
-        io.filewrite(utils.unmask(betas, mask),
-                     op.join(out_dir, '{0}betas_catd.nii'.format(label)),
-                     ref_img)
+        for i_echo in range(n_echos):
+            echo_betas = betas[:, i_echo, :]
+            io.filewrite(
+                utils.unmask(echo_betas, mask),
+                op.join(out_dir, 'echo-{0}_desc-{1}_components.nii.gz'.format(
+                    i_echo + 1, label)),
+                ref_img
+            )
 
         # Echo-specific maps of predicted values for R2 and S0 models for each
         # component.
-        io.filewrite(utils.unmask(pred_R2_maps, mask),
-                     op.join(out_dir, '{0}R2_pred.nii'.format(label)), ref_img)
-        io.filewrite(utils.unmask(pred_S0_maps, mask),
-                     op.join(out_dir, '{0}S0_pred.nii'.format(label)), ref_img)
+        io.filewrite(
+            utils.unmask(pred_R2_maps, mask),
+            op.join(out_dir, 'desc-{0}R2ModelPredictions_X.nii.gz'.format(label)),
```

Comment:
X is not a valid suffix.

Reply:
This may not matter as much since I'm dropping these intermediate images in #475.